Add lawn mower entity (#1875)
home-assistant/developers.home-assistant: docs/core/entity/lawn-mower.md

---
title: Lawn Mower Entity
sidebar_label: Lawn Mower
---

Derive entity platforms from [`homeassistant.components.lawn_mower.LawnMowerEntity`](https://github.com/home-assistant/home-assistant/blob/master/homeassistant/components/lawn_mower/__init__.py)

## Properties

:::tip
Properties should always only return information from memory and not do I/O (like network requests). Implement `update()` or `async_update()` to fetch data.
:::

| Name | Type | Default | Description
| ---- | ---- | ------- | -----------

> | name | string | **Required** | Name of the entity.

Reviewer: ```suggestion
```

> | state | string | **Required** | One of the states listed in the states section.

Reviewer: ```suggestion
| activity | LawnMowerActivity | **Required** | One of the activities in the LawnMowerActivity enum.
```

| supported_features | int | (abstract method) | Bitmap of supported features. See below.

> ## States

Reviewer: ```suggestion
## Activites
```

> | State | Description

Reviewer: ```suggestion
| Activity | Description
```

| ----- | -----------
| `MOWING` | The lawn mower is currently mowing.
| `DOCKED` | The lawn mower is done mowing and is currently docked.
| `PAUSED` | The lawn mower was active and is now paused.
| `ERROR` | The lawn mower encountered an error while active and needs assistance.

## Supported Features

Supported features are defined by using values in the `LawnMowerEntityFeature` enum
and are combined using the bitwise or (`|`) operator.

> Note that all lawn mower entity platforms derived from `homeassistant.components.lawn_mower.LawnMowerEntity`

Reviewer: Remove the last sentence. It's not applicable.
